### PR TITLE
mold: 0.9.3 -> 0.9.6

### PR DESCRIPTION
**Pull request by jakobrs**
###### Motivation for this change

Version update: https://github.com/NixOS/nixpkgs/pull/131830#issuecomment-930144248.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
I guess this will need some darwin testing? or would we drop darwin from platforms?

> Note that mold 0.9.4 can't be built on Darwin per se, and even if you can, it can't be used to create macOS binaries. It can only create ELF (Linux) binaries. So you can use it for cross compilation but not for macOS native development.

https://github.com/NixOS/nixpkgs/pull/131830#issuecomment-913324672

**Review by nitsky — APPROVED** [review]

File: pkgs/development/tools/mold/default.nix [hunk 1]
```diff
@@ -12,23 +12,15 @@
...
     owner = "rui314";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256:1z9i8nvdl9h0zydh1gd9244q96n9x1gh5y90m71bghnh7nws0zmd";
+    sha256 = "sha256:0mj258fy8l4i23jd6ail0xrrq3das7lmrf1brrr1591ahx4vjj14";
   };
 
-  patches = [
-    # Intercept all invocations of ld, ld.gold or ld.lld
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/mold/default.nix:26]: Please remove the unused fetchpatch from inputs.

Comment (jakobrs) [reply pkgs/development/tools/mold/default.nix:26]: Completely missed that, sorry.

**Comment by 06kellyjac** [comment]
Failed to build on my MacOS VM

```
In file included from third-party/tbb/include/tbb/../oneapi/tbb/task.h:22:
third-party/tbb/include/tbb/../oneapi/tbb/detail/_task.h:221:13: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
    virtual ~task() = default;
            ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: in defaulted destructor for 'tbb::detail::d1::task' first required here
struct start_for : public task {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: in implicit destructor for 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:72:49), long long>, const tbb::detail::d1::auto_partitioner>' first required here
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:247:5: note: in instantiation of template class 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:72:49), long long>, const tbb::detail::d1::auto_partitioner>' requested here
    start_for<Range,Body,const auto_partitioner>::run(range,body,partitioner);
    ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:316:9: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:72:49), long long>>' requested here
        parallel_for(range, body, partitioner);
        ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:355:5: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for_impl<long long, (lambda at elf/compress.cc:72:49), const tbb::detail::d1::auto_partitioner>' requested here
    parallel_for_impl<Index,Function,const auto_partitioner>(first, last, static_cast<Index>(1), f, auto_partitioner());
    ^
elf/compress.cc:72:8: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<long long, (lambda at elf/compress.cc:72:49)>' requested here
  tbb::parallel_for((i64)0, (i64)inputs.size(), [&](i64 i) {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/detail/_task.h:221:13: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
    virtual ~task() = default;
            ^
In file included from elf/compress.cc:17:
In file included from third-party/tbb/include/tbb/parallel_for_each.h:17:
In file included from third-party/tbb/include/tbb/../oneapi/tbb/parallel_for_each.h:27:
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
struct start_for : public task {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: in implicit destructor for 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:72:49), long long>, const tbb::detail::d1::auto_partitioner>' first required here
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:247:5: note: in instantiation of template class 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:72:49), long long>, const tbb::detail::d1::auto_partitioner>' requested here
    start_for<Range,Body,const auto_partitioner>::run(range,body,partitioner);
    ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:316:9: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:72:49), long long>>' requested here
        parallel_for(range, body, partitioner);
        ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:355:5: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for_impl<long long, (lambda at elf/compress.cc:72:49), const tbb::detail::d1::auto_partitioner>' requested here
    parallel_for_impl<Index,Function,const auto_partitioner>(first, last, static_cast<Index>(1), f, auto_partitioner());
    ^
elf/compress.cc:72:8: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<long long, (lambda at elf/compress.cc:72:49)>' requested here
  tbb::parallel_for((i64)0, (i64)inputs.size(), [&](i64 i) {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
struct start_for : public task {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
struct start_for : public task {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: in implicit destructor for 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:101:49), long long>, const tbb::detail::d1::auto_partitioner>' first required here
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:247:5: note: in instantiation of template class 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:101:49), long long>, const tbb::detail::d1::auto_partitioner>' requested here
    start_for<Range,Body,const auto_partitioner>::run(range,body,partitioner);
    ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:316:9: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:101:49), long long>>' requested here
        parallel_for(range, body, partitioner);
        ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:355:5: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for_impl<long long, (lambda at elf/compress.cc:101:49), const tbb::detail::d1::auto_partitioner>' requested here
    parallel_for_impl<Index,Function,const auto_partitioner>(first, last, static_cast<Index>(1), f, auto_partitioner());
    ^
elf/compress.cc:101:8: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<long long, (lambda at elf/compress.cc:101:49)>' requested here
  tbb::parallel_for((i64)0, (i64)shards.size(), [&](i64 i) {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
struct start_for : public task {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
struct start_for : public task {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: in implicit destructor for 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:120:49), long long>, const tbb::detail::d1::auto_partitioner>' first required here
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:247:5: note: in instantiation of template class 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:120:49), long long>, const tbb::detail::d1::auto_partitioner>' requested here
    start_for<Range,Body,const auto_partitioner>::run(range,body,partitioner);
    ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:316:9: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:120:49), long long>>' requested here
        parallel_for(range, body, partitioner);
        ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:355:5: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for_impl<long long, (lambda at elf/compress.cc:120:49), const tbb::detail::d1::auto_partitioner>' requested here
    parallel_for_impl<Index,Function,const auto_partitioner>(first, last, static_cast<Index>(1), f, auto_partitioner());
    ^
elf/compress.cc:120:8: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<long long, (lambda at elf/compress.cc:120:49)>' requested here
  tbb::parallel_for((i64)0, (i64)inputs.size(), [&](i64 i) {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
struct start_for : public task {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
struct start_for : public task {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: in implicit destructor for 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:154:49), long long>, const tbb::detail::d1::auto_partitioner>' first required here
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:247:5: note: in instantiation of template class 'tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:154:49), long long>, const tbb::detail::d1::auto_partitioner>' requested here
    start_for<Range,Body,const auto_partitioner>::run(range,body,partitioner);
    ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:316:9: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<tbb::detail::d1::blocked_range<long long>, tbb::detail::d1::parallel_for_body_wrapper<(lambda at elf/compress.cc:154:49), long long>>' requested here
        parallel_for(range, body, partitioner);
        ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:355:5: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for_impl<long long, (lambda at elf/compress.cc:154:49), const tbb::detail::d1::auto_partitioner>' requested here
    parallel_for_impl<Index,Function,const auto_partitioner>(first, last, static_cast<Index>(1), f, auto_partitioner());
    ^
elf/compress.cc:154:8: note: in instantiation of function template specialization 'tbb::detail::d1::parallel_for<long long, (lambda at elf/compress.cc:154:49)>' requested here
  tbb::parallel_for((i64)0, (i64)shards.size(), [&](i64 i) {
       ^
third-party/tbb/include/tbb/../oneapi/tbb/parallel_for.h:66:8: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
struct start_for : public task {
       ^
5 errors generated.
make: *** [<builtin>: elf/compress.o] Error 1
builder for '/nix/store/kmd5j72dmfxk3pb4aj9by1p033s8nd66-mold-0.9.6.drv' failed with exit code 2
error: build of '/nix/store/kmd5j72dmfxk3pb4aj9by1p033s8nd66-mold-0.9.6.drv' failed
```

File: pkgs/development/tools/mold/default.nix [hunk 2]
```diff
@@ -12,23 +11,15 @@
...
   pname = "mold";
-  version = "0.9.3";
+  version = "0.9.6";
 
   src = fetchFromGitHub {
     owner = "rui314";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256:1z9i8nvdl9h0zydh1gd9244q96n9x1gh5y90m71bghnh7nws0zmd";
+    sha256 = "sha256:0mj258fy8l4i23jd6ail0xrrq3das7lmrf1brrr1591ahx4vjj14";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/mold/default.nix:20]:
```suggestion
    sha256 = "0mj258fy8l4i23jd6ail0xrrq3das7lmrf1brrr1591ahx4vjj14";
```

**Pull request merged**